lib/views/auth/auth_page_form.dart

```diff
@@ -0,0 +1,298 @@
+      child: LenraFlex(
+        spacing: 24,
+        direction: Axis.vertical,
+        crossAxisAlignment: CrossAxisAlignment.center,
+        children: [
+          LenraTextFormField(
+            label: 'Your email',
+            hintText: 'Email',
+            onChanged: (String value) {
+              email = value;
+            },
+            onSubmitted: (_) {
+              submit();
+            },
+            type: LenraTextFormFieldType.email,
+            size: LenraComponentSize.large,
+            autofillHints: [AutofillHints.email],
+          ),
+          //------Password------
+          LenraTextFormField(
+            label: 'Your password',
+            obscure: _hidePassword,
+            hintText: 'Password',
+            onChanged: (String value) {
+              password = value;
+            },
+            onSubmitted: (_) {
+              submit();
+            },
+            type: LenraTextFormFieldType.password,
+            size: LenraComponentSize.large,
+            autofillHints: [AutofillHints.password],
+            onSuffixPressed: () {
+              setState(() {
+                _hidePassword = !_hidePassword;
+              });
+            },
+          ),
+          LenraFlex(
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: [
+              LenraCheckbox(
+                value: keep,
+                onPressed: (value) {
+                  setState(() {
+                    keep = value!;
+                  });
+                },
+              ),
+              LenraText(
+                text: "Keep me logged in",
+                style: themeData.lenraTextThemeData.disabledBodyText,
+              ),
+            ],
+          ),
+          LenraFlex(
+            direction: Axis.vertical,
+            crossAxisAlignment: CrossAxisAlignment.center,
+            children: [
+              LoadingButton(
+                text: "Log in",
+                onPressed: () {
+                  submit();
+                },
+                loading: isLogging,
+              ),
+              Container(
+                padding: EdgeInsets.only(top: 10.0),
+                child: RichText(
+                  text: TextSpan(
+                    text: "I forgot my password",
+                    style: themeData.lenraTextThemeData.blueBodyText,
+                    recognizer: TapGestureRecognizer()
+                      ..onTap = () {
+                        CommonNavigator.go(context, CommonNavigator.lostPassword);
+                      },
+                  ),
+                ),
+              ),
+            ],
+          ),
+        ],
+      ),
+    );
+  }
+
+  void submit() {
+    if (_formKey.currentState!.validate()) {
+      AuthModel authModel = context.read<AuthModel>();
+      if (isRegisterPage) {
+        authModel.register(email, password).then((_) {
+          GoRouter.of(context).go(context.read<AuthModel>().redirectToRoute ?? '/');
```

Comment:
What happens when the e-mail is not verified (after the registration for example) ?

Reply:
If you did not accept the CGU, you are redirected to it.
If you accepted the CGU and did not validate the account, you are redirected to the validation-user page

Reply:
The problem then is that the redirection is not done properly to the context.read<AuthModel>().redirectToRoute because the CGU and validation do not use it. I can make them use it.

Reply:
EDIT: I cannot make them use the redirectToRoute value. We would need to rethink the login/register process as mentioned by @Nesqwik on Discord.